flow/BlobCipher.cpp

```diff
@@ -0,0 +1,387 @@
+/*
+ * BlobCipher.cpp
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2022 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "flow/BlobCipher.h"
+#include "flow/Error.h"
+#include "flow/FastRef.h"
+#include "flow/IRandom.h"
+#include "flow/network.h"
+#include "flow/Trace.h"
+#include "flow/UnitTest.h"
+#include <cstring>
+#include <memory>
+
+//#if ENCRYPTION_ENABLED
+
+uint64_t BlobCipherKeyCache::CIPHER_KEY_CACHE_TTL_NS = 10 * 60 * 1000 * 1000 * 1000L;
+uint8_t EncryptBlobCipher::ENCRYPT_HEADER_VERSION = 1;
```

Comment:
```suggestion
static constexpr uint64_t BlobCipherKeyCache::CIPHER_KEY_CACHE_TTL_NS = 10 * 60 * 1000 * 1000 * 1000L;
static constexpr uint8_t EncryptBlobCipher::ENCRYPT_HEADER_VERSION = 1;
```

Reply:
For clarification: I think recommendation above is to initialize "static variable" inside the class (along with declaration) such as:

```
class BlobCipherKeyCache : NonCopyable {
	static constexpr uint64_t CIPHER_KEY_CACHE_TTL_NS;
```

Above suggestion is quoted on a .cpp file where compiler would be unhappy using "static" keyword, hence, the clarification:

```
foundationdb/flow/BlobCipher.h:177:28: error: default initialization of an object of const type 'const uint64_t' (aka 'const unsigned long')
        static constexpr uint64_t CIPHER_KEY_CACHE_TTL_NS;
```

Thanks